src/cmd-buildextend-metal

```diff
@@ -139,24 +139,37 @@ fi
 imgname=${name}-${build}-${image_type}.${basearch}.${image_format}
 imgpath=${PWD}/${imgname}
 
-deploy_via_container=""
-if jq -re '.["deploy-via-container"]' < "${image_json}"; then
-    deploy_via_container="true"
-fi
+# Parse the passed config JSON and extract a mandatory value
+getconfig() {
+    k=$1
+    config=$2
+    jq -re .\""$k"\" < "${config}"
+}
+# Return a configuration value, or default if not set
+getconfig_def() {
+    k=$1
+    shift
+    default=$1
+    config=$2
+    jq -re .\""$k"\"//\""${default}"\" < "${config}"
+}
+
+# Grab a few values from ${image_json}
+deploy_via_container=$(getconfig_def "deploy-via-container" "" "${image_json}")
```

Comment:
We are removing the `deploy_via_container=""` what will be the return if we don't have it in the image_json, it maybe break the podman build

Reply:
```
getconfig_def "deploy-via-container" "" "${image_json}"
```

The function sets a default value to `""` (which we are passing here as the 2nd argument) if there is no value for it in the image.json.